.conan-recipes/llnl-units/conanfile.py

```diff
@@ -0,0 +1,79 @@
+from conans import ConanFile, CMake, tools
+import os
+
+CMAKE_PROJECT_STR = """project(
+    UNITS
+    LANGUAGES CXX
+    VERSION 0.5.0
+)"""
+
+
+class UnitsConan(ConanFile):
+    name = "LLNL-Units"
+    version = "0.5.0"
+    license = "BSD-3"
+    url = "https://github.com/scipp/conan-llnl-units"
+    homepage = "https://units.readthedocs.io"
+    description = ("A run-time C++ library for working with units "
+                   "of measurement and conversions between them "
+                   "and with string representations of units "
+                   "and measurements")
+    topics = ("units", "dimensions", "quantities", "physical-units",
+              "dimensional-analysis", "run-time")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "base_type": ["uint32_t", "uint64_t"],
+        "namespace": "ANY"
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "base_type": "uint32_t",
+        "namespace": None
+    }
+    generators = "cmake"
+
+    def source(self):
+        git = tools.Git("units")
+        git.clone("https://github.com/SimonHeybrock/units.git")
+
+        cmake_project_str = (CMAKE_PROJECT_STR.replace("\n", os.linesep)
+                             if self.settings.os == "Windows" else CMAKE_PROJECT_STR)
+
+        tools.replace_in_file(
+            "units/CMakeLists.txt", cmake_project_str, cmake_project_str + """
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()""")
+
+    def build(self):
+        cmake = CMake(self)
+        units_namespace = self.options.get_safe("namespace")
+        cmake.definitions["UNITS_ENABLE_TESTS"] = "OFF"
+        cmake.definitions["UNITS_BASE_TYPE"] = self.options.base_type
+        if units_namespace:
+            cmake.definitions["UNITS_NAMESPACE"] = units_namespace
+        if self.options["shared"]:
+            cmake.definitions["UNITS_BUILD_SHARED_LIBRARY"] = "ON"
+            cmake.definitions["UNITS_BUILD_STATIC_LIBRARY"] = "OFF"
+        # The library uses C++14, but we want to set the namespace
+        # to llnl::units which requires C++17.
```

Comment:
Out of interest - why do we need c++17 to customise the namespace?

Reply:
It is just about the `::` for nested namespaces. The original code might have something like:
```cpp
NAMESPACE namespace {
}
```
where `NAMESPACE` is set by an define. Pre C++17 you'd have to write
```cpp
namespace llnl {
namespace units {
}
}
```
which is not possible using a simple define.

Reply:
Ah yes, I was vaguely aware of that feature. Thanks for explaining.